File: public/usage-examples/physics/matrix_multiply_matrix/matrix_multiply_matrix-simple.py
from splashkit import *

# Define and populate the matrix
my_matrix_1 = Matrix2D()
my_matrix_1.elements[0][0] = 1
my_matrix_1.elements[0][1] = 2
my_matrix_1.elements[0][2] = 3

my_matrix_1.elements[1][0] = 0
my_matrix_1.elements[1][1] = 1
my_matrix_1.elements[1][2] = 4

my_matrix_1.elements[2][0] = 5
my_matrix_1.elements[2][1] = 6
my_matrix_1.elements[2][2] = 0

# Define the point
my_point_1 = Point2D()
my_point_1.x = 200
my_point_1.y = 100

# Multiply the matrix by the point
result_point = matrix_multiply(my_matrix_1, my_point_1)

# Print the matrix and points
write_line(matrix_to_string(my_matrix_1))
write_line(point_to_string(my_point_1))
write_line(point_to_string(result_point))

File: public/usage-examples/physics/matrix_multiply_matrix/matrix_multiply_matrix-simple.cpp
#include "splashkit.h"

int main()
{
    // Define and populate the first matrix
    matrix_2d my_matrix_1;
    my_matrix_1.elements[0][0] = 1;
    my_matrix_1.elements[0][1] = 2;
    my_matrix_1.elements[0][2] = 3;

    my_matrix_1.elements[1][0] = 0;
    my_matrix_1.elements[1][1] = 1;
    my_matrix_1.elements[1][2] = 4;

    my_matrix_1.elements[2][0] = 5;
    my_matrix_1.elements[2][1] = 6;
    my_matrix_1.elements[2][2] = 0;

    // Define and populate the second matrix
    matrix_2d my_matrix_2;
    my_matrix_2.elements[0][0] = 7;
    my_matrix_2.elements[0][1] = 8;
    my_matrix_2.elements[0][2] = 9;

    my_matrix_2.elements[1][0] = 1;
    my_matrix_2.elements[1][1] = 0;
    my_matrix_2.elements[1][2] = 2;

    my_matrix_2.elements[2][0] = 3;
    my_matrix_2.elements[2][1] = 4;
    my_matrix_2.elements[2][2] = 5;

    // Multiply the two matrices
    matrix_2d result_matrix = matrix_multiply(my_matrix_1, my_matrix_2);

    // Print the original matrices and the result
    write_line("Matrix 1:");
    write_line(matrix_to_string(my_matrix_1));

    write_line("Matrix 2:");
    write_line(matrix_to_string(my_matrix_2));

    write_line("Resulting Matrix (Matrix 1 x Matrix 2):");
    write_line(matrix_to_string(result_matrix));

    return 0;
}
